StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+API.swift

```diff
@@ -591,75 +626,87 @@ extension PaymentSheet {
+                let clientSecret = try await fetchIntentClientSecretFromMerchant(intentConfig: deferredIntentContext.intentConfig,
+                                                                                 paymentMethodID: _paymentMethod.stripeId,
+                                                                                 shouldSavePaymentMethod: shouldSavePaymentMethod)
+                // Finish confirmation
+                finishDeferredConfirmation(deferredIntentContext: deferredIntentContext, clientSecret: clientSecret)
+
+            } catch {
+                deferredIntentContext.completion(.failed(error: error))
             }
+        }
+    }
 
-            guard let paymentMethod = paymentMethod else {
-                let paymentMethodCreationError = PaymentSheetError.unknown(
-                    debugDescription: "Failed to create a new payment method."
-                )
-                completion(.failed(error: paymentMethodCreationError))
-                return
-            }
+    static func fetchIntentClientSecretFromMerchant(intentConfig: IntentConfiguration,
+                                                    paymentMethodID: String,
+                                                    shouldSavePaymentMethod: Bool) async throws -> String {
+      try await withCheckedThrowingContinuation { continuation in
+
+          if let confirmHandlerForServerSideConfirmation = intentConfig.confirmHandlerForServerSideConfirmation {
+              confirmHandlerForServerSideConfirmation(paymentMethodID, shouldSavePaymentMethod, { result in
+                  continuation.resume(with: result)
+              })
+          } else if let confirmHandler = intentConfig.confirmHandler {
+              confirmHandler(paymentMethodID, { result in
+                  continuation.resume(with: result)
+              })
+          }
+      }
+    }
 
-            if let serverConfirmHandler = intentConfig.confirmHandlerForServerSideConfirmation {
-                serverConfirmHandler(paymentMethod.stripeId,
-                                            shouldSavePaymentMethod,
-                                            _handleIntentCreation(_:))
-            } else if let confirmHandler = intentConfig.confirmHandler {
-                confirmHandler(paymentMethod.stripeId, _handleIntentCreation(_:))
-            } else {
-                fatalError("Either confirmHandler or confirmHandlerForServerSideConfirmation must be non-nil when using deferred intents")
-            }
+    static func finishDeferredConfirmation(deferredIntentContext: DeferredIntentContext,
+                                           clientSecret: String) {
+        if deferredIntentContext.isServerSideConfirmation {
+            handleServerConfirmedIntent(deferredContext: deferredIntentContext, clientSecret: clientSecret)
+        } else {
+            confirmClientSide(for: deferredIntentContext, with: clientSecret)
         }
     }
```

Comment:
I'm not seeing the value in extracting this code into a helper method.  It can only be called from exactly one specific place in our code, right - as step 3 of `handleDeferred..`

We should inline this if so.

Reply:
Same applies to the inner `handleServer..` and `confirmClientSide`.. methods.  Can we inline it all? I don't want to pass control over to other helper methods - it's confusing to follow. 

We can extract work out of the 'master' method into helper methods, but if we do so they should return something and be callable from any point in the code.